guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/test/java/org/guvnor/asset/management/client/editors/repository/wizard/pages/RepositoryInfoPageTest.java

```diff
@@ -74,24 +90,36 @@ public void testPageLoad() {
 
         verify( view, times( 1 ) ).init( infoPage );
         verify( view ).initOrganizationalUnits( eq( organizationalUnitsInfo ) );
+        verify( statusChangeHandler, never() ).handleEvent( any( WizardPageStatusChangeEvent.class ) );
 
         assertPageComplete( false, infoPage );
     }
 
     /**
-     * Tests that the page reacts properly when the selected organizational unit changes.
+     * Tests that the page reacts properly when there is no selected organizational unit.
      */
     @Test
-    public void testOrganizationalUnitChange() {
+    public void testNoSelectedOrganizationalUnit() {
 
-        RepositoryInfoPage infoPage = new RepositoryInfoPageExtended( view,
-                new OrganizationalUnitServiceCallerMock( organizationalUnitService ),
-                new RepositoryServiceCallerMock( repositoryService ),
-                true,
-                new WizardTestUtils.WizardPageStatusChangeEventMock() );
+        when( organizationalUnitService.getOrganizationalUnits() ).thenReturn( organizationalUnits );
+        when( view.getOrganizationalUnitName() ).thenReturn( RepositoryInfoPageView.NOT_SELECTED );
 
-        CreateRepositoryWizardModel model = new CreateRepositoryWizardModel();
-        infoPage.setModel( model );
+        infoPage.prepareView();
+        infoPage.onOUChange();
+
+        verify( view, times( 1 ) ).getOrganizationalUnitName();
+        verify( statusChangeHandler, never() ).handleEvent( any( WizardPageStatusChangeEvent.class ) );
```

Comment:
Does the ```WizardPageStatusChangeEvent``` not get called whenever something changes on the page? I'd have thought selecting ```RepositoryInfoPageView.NOT_SELECTED``` would still trigger a status event to be triggered?